tests/Unit/Parallel/Test_InboxInserters.cpp

```diff
@@ -0,0 +1,395 @@
+                    const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    Parallel::receive_data<Tags::PushbackTag>(
+        Parallel::get_parallel_component<ParallelComponent>(cache)[0], 1_st,
+        23);
+    db::mutate<Tags::PushbackCounter>(make_not_null(&box), [
+    ](const gsl::not_null<size_t*> pushback_counter) noexcept {
+      (*pushback_counter)++;
+    });
+    return std::forward_as_tuple(std::move(box));
+  }
+};
+
+struct ReceivePushback {
+  using inbox_tags = tmpl::list<Tags::PushbackTag>;
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    db::mutate<Tags::PushbackCounter>(make_not_null(&box), [
+    ](const gsl::not_null<size_t*> pushback_counter) noexcept {
+      (*pushback_counter)++;
+    });
+    return std::forward_as_tuple(std::move(box));
+  }
+
+  template <typename DbTags, typename Metavariables, typename... InboxTags,
+            typename ArrayIndex>
+  static bool is_ready(
+      const db::DataBox<DbTags>& /*box*/,
+      const tuples::TaggedTuple<InboxTags...>& inboxes,
+      const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/) noexcept {
+    return get<Tags::PushbackTag>(inboxes).at(1_st).size() == 1;
+  }
+};
+}  // namespace Actions
+
+template <typename Metavariables>
+struct Component {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<
+              db::AddSimpleTags<Tags::MapCounter, Tags::MemberInsertCounter,
+                                Tags::ValueCounter, Tags::PushbackCounter>>>>,
+
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Testing,
+          tmpl::list<Actions::SendMap, Actions::ReceiveMap,
+                     Actions::SendMemberInsert, Actions::ReceiveMemberInsert,
+                     Actions::SendValue, Actions::ReceiveValue,
+                     Actions::SendPushback, Actions::ReceivePushback>>>;
+};
+
+struct Metavariables {
+  using component_list = tmpl::list<Component<Metavariables>>;
+
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+SPECTRE_TEST_CASE("Unit.Parallel.InboxInserters", "[Unit]") {
```

Comment:
other tests in this directory are labelled [Parallel].

Reply:
Thanks :) Added